src/ERC20Reserve.sol

```diff
@@ -5,54 +5,49 @@ import {Ownable} from "openzeppelin-contracts/access/Ownable.sol";
 import {IERC20} from "openzeppelin-contracts/token/ERC20/IERC20.sol";
 
 interface IERC20Reserve {
-    function erc20() external view returns (IERC20);
+    function withdrawable(IERC20 token) external view returns (uint256 amt);
 
-    function withdraw(uint256 amt) external;
+    function withdraw(IERC20 token, uint256 amt) external;
 
-    function deposit(uint256 amt) external;
+    function deposit(IERC20 token, uint256 amt) external;
 }
 
 contract ERC20Reserve is IERC20Reserve, Ownable {
-    IERC20 public immutable override erc20;
     address public user;
```

Comment:
I think i still would prefer a mapping for the `user`. It would just give us more potential freedom, in case we end up with a legacy version which still wants to have access to the reserve and a newer one etc.

Reply:
I agree, fixed